framework/src/task_manager/task_manager_set_callback.c

```diff
@@ -293,15 +293,19 @@ int task_manager_set_broadcast_cb(int msg, tm_broadcast_callback_t func, tm_msg_
 
 	TM_ASPRINTF(&request_msg.q_name, "%s%d", TM_PRIVATE_MQ, request_msg.caller_pid);
 	if (request_msg.q_name == NULL) {
-		TM_FREE(((tm_msg_t *)((tm_broadcast_info_t *)request_msg.data)->cb_data)->msg);
-		TM_FREE(((tm_broadcast_info_t *)request_msg.data)->cb_data);
+		if (((tm_broadcast_info_t *)request_msg.data)->cb_data != NULL) {
+			TM_FREE(((tm_msg_t *)((tm_broadcast_info_t *)request_msg.data)->cb_data)->msg);
```

Comment:
If TM_FREE macro can have the null check we dont need to add this check everywhere.

Reply:
That's a good idea. I will modify this with another PR.
Thanks!